File: apps/android/core/data/src/main/kotlin/app/gauja/core/data/servers/ServerMapper.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.data.servers

import app.gauja.core.api.models.GetStatus200Response
import app.gauja.core.api.models.PublicSettings
import app.gauja.core.compat.ServerVersion
import app.gauja.core.model.Compatibility
import app.gauja.core.model.MediaServerType
import app.gauja.core.model.ServerAddress
import app.gauja.core.model.ServerSnapshot

internal fun mapServer(
    address: ServerAddress,
    status: GetStatus200Response,
    settings: PublicSettings,
): ServerSnapshot =
    ServerSnapshot(
        address,
        status.version,
        settings.applicationTitle,
        settings.initialized,
        status.restartRequired,
        settings.localLogin,
        settings.mediaServerLogin,
        when (settings.mediaServerType) {
            1.0 -> MediaServerType.PLEX
            2.0 -> MediaServerType.JELLYFIN
            3.0 -> MediaServerType.EMBY
            4.0 -> MediaServerType.NOT_CONFIGURED
            else -> MediaServerType.UNKNOWN
        },
        ServerVersion.parse(status.version)?.compatibility() ?: Compatibility.UNKNOWN,
    )

File: apps/ios/Packages/Data/Sources/Data/servers/ServerMapper.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
import Compat
import Model
internal import SeerrAPI

func mapServer(
    _ address: ServerAddress, _ status: Operations.getStatus.Output.Ok.Body.jsonPayload,
    _ settings: Components.Schemas.PublicSettings
) -> ServerSnapshot {
    let mediaServerType: MediaServerType
    switch settings.mediaServerType {
    case 1: mediaServerType = .plex
    case 2: mediaServerType = .jellyfin
    case 3: mediaServerType = .emby
    case 4: mediaServerType = .notConfigured
    default: mediaServerType = .unknown
    }
    return ServerSnapshot(
        address: address, version: status.version, title: settings.applicationTitle,
        initialized: settings.initialized, restartRequired: status.restartRequired,
        localLogin: settings.localLogin, mediaServerLogin: settings.mediaServerLogin,
        mediaServerType: mediaServerType,
        compatibility: ServerVersion(status.version)?.compatibility ?? .unknown)
}
